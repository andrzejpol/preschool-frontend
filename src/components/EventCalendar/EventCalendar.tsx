import {useState} from "react";
import Calendar from "react-calendar";
import 'react-calendar/dist/Calendar.css';
import {faCalendarDays} from "@fortawesome/free-solid-svg-icons";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

type ValuePiece = Date | null;
type Value = ValuePiece | [ValuePiece, ValuePiece];

const events = [
  {
    id: 1,
    title: "Lorem ipsum dolor",
    time: "3:00PM - 7:00PM",
    description: "Lorem ipsum dolor sit amet"
  },
  {
    id: 2,
    title: "Lorem ipsum dolor",
    time: "12:00AM - 2:00AM",
    description: "Lorem ipsum dolor sit amet"
  },
  {
    id: 3,
    title: "Lorem ipsum dolor",
    time: "12:00PM - 2:00PM",
    description: "Lorem ipsum dolor sit amet"
  },
]

const EventCalendar = () => {
  const [value, onChange] = useState<Value>(new Date());

  return (
    <div className="bg-white p-4 rounded-md">
      <Calendar onChange={onChange} value={value} locale="en"/>
      <div className="flex items-center justify-between">
        <h1 className="text-xl font-semibold my-4">Events</h1>
        <FontAwesomeIcon icon={faCalendarDays} className="w-[20px] h-[20px]"/>
      </div>
      <div className="flex flex-col gap-4">
        {events.map(event => (
          <div
            className="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
            key={event.id}>
            <div className="flex items-center justify-between">
              <h1 className="font-semibold text-gray-600">{event.title}</h1>
              <span className="text-preschoolPrimary-300 text-xs">{event.time}</span>
            </div>
            <p className="mt-2 text-gray-400 text-sm">{event.description}</p>
          </div>
        ))}
      </div>
    </div>
  )
};

export default EventCalendar;