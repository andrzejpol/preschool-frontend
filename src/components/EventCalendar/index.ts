import EventCalendar from "./EventCalendar.tsx";


export default EventCalendar;