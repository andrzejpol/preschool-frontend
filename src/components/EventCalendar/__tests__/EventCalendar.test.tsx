import React from "react";
import {render} from "@testing-library/react";
import EventCalendar from "../EventCalendar";

vi.mock("react-calendar", () => ({
  default: ({onChange, value}) => (
    <div>
      Mocked Calendar Component - Selected Date: {value.toString()}
      <button onClick={() => onChange(new Date())}>Change Date</button>
    </div>
  ),
}));

describe("EventCalendar Component", () => {
  it("matches the snapshot", () => {
    const {asFragment} = render(<EventCalendar/>);

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });
});
