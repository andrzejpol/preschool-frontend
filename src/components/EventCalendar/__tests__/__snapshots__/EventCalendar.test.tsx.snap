// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EventCalendar Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-4 rounded-md"
  >
    <div>
      Mocked Calendar Component - Selected Date: Sun Jan 26 2025 04:16:53 GMT+0100 (Central European Standard Time)
      <button>
        Change Date
      </button>
    </div>
    <div
      class="flex items-center justify-between"
    >
      <h1
        class="text-xl font-semibold my-4"
      >
        Events
      </h1>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-calendar-days w-[20px] h-[20px]"
        data-icon="calendar-days"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 0c17.7 0 32 14.3 32 32l0 32 128 0 0-32c0-17.7 14.3-32 32-32s32 14.3 32 32l0 32 48 0c26.5 0 48 21.5 48 48l0 48L0 160l0-48C0 85.5 21.5 64 48 64l48 0 0-32c0-17.7 14.3-32 32-32zM0 192l448 0 0 272c0 26.5-21.5 48-48 48L48 512c-26.5 0-48-21.5-48-48L0 192zm64 80l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm128 0l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0zM64 400l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0zm112 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="flex flex-col gap-4"
    >
      <div
        class="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
      >
        <div
          class="flex items-center justify-between"
        >
          <h1
            class="font-semibold text-gray-600"
          >
            Lorem ipsum dolor
          </h1>
          <span
            class="text-preschoolPrimary-300 text-xs"
          >
            3:00PM - 7:00PM
          </span>
        </div>
        <p
          class="mt-2 text-gray-400 text-sm"
        >
          Lorem ipsum dolor sit amet
        </p>
      </div>
      <div
        class="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
      >
        <div
          class="flex items-center justify-between"
        >
          <h1
            class="font-semibold text-gray-600"
          >
            Lorem ipsum dolor
          </h1>
          <span
            class="text-preschoolPrimary-300 text-xs"
          >
            12:00AM - 2:00AM
          </span>
        </div>
        <p
          class="mt-2 text-gray-400 text-sm"
        >
          Lorem ipsum dolor sit amet
        </p>
      </div>
      <div
        class="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
      >
        <div
          class="flex items-center justify-between"
        >
          <h1
            class="font-semibold text-gray-600"
          >
            Lorem ipsum dolor
          </h1>
          <span
            class="text-preschoolPrimary-300 text-xs"
          >
            12:00PM - 2:00PM
          </span>
        </div>
        <p
          class="mt-2 text-gray-400 text-sm"
        >
          Lorem ipsum dolor sit amet
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
